# テーマ1 Python基礎演習

## 副題
Notebook の使い方と、データ分析に入る前の Python 準備

## 対象
Python を初めて学ぶ大学1年生

## この回の位置づけ
この回ではグラフは扱いません。可視化はテーマ2でまとめて扱います。
まずは Python を書いて実行し、結果を見て、少し直して、また動かすことに慣れてもらいます。
データ分析などで必要となるNumPy の配列・行列も最後に少しだけ触れます。

## 到達目標
- Notebook 上でセルを実行し、結果を確認しながら試行錯誤できる
- 加減乗除、変数、`print`、文字列、リスト、条件分岐、繰り返し、関数の基本を使える
- NumPy 配列の形と簡単な集計を確認できる
- 次回以降で使う Iris データを読み込み、表として中身を確認できる

## 今日の成果物
- 数値や文字列を使った簡単な Python コードを自分で書き換えて実行する
- 関数を1つ使って計算結果を確認する
- NumPy の2次元配列の行数・列数・平均を確認する
- Iris データの先頭行と列名を確認する

## 進め方
- 上から順にセルを実行してください
- `TODO` の部分を少しずつ書き換えながら進めてください
- わからないところは、まず実行結果を見てから前のセルに戻って確認してください

## この回で大切にする考え方

Python の文法を最初から全部覚える必要はありません。この授業では、次の小さなサイクルを何度も回します。

1. コードを実行する
2. 結果を見る
3. 予想と違うところを直す
4. 結果を自分の言葉で説明する

Notebook は、このサイクルを短く回すための道具です。セルを実行したら、出力を必ず見て「何が起きたか」を確認してください。


### うまく動かないときの見方

エラーが出たときは、まず最後の1行に注目してください。Python のエラーは長く見えますが、多くの場合、最後の行に「何が問題だったか」が短く書かれています。

よくある例:

- `NameError`: 変数名をまだ作っていない、またはスペルが違う
- `SyntaxError`: `:` や `)` など、書き方の記号が足りない
- `IndentationError`: `if` や `for` の中の字下げがずれている

エラーは失敗ではなく、コンピュータからのヒントです。直したら同じセルをもう一度実行してください。


### Step 0: 加減乗除の確認
まずは Python の基本演算子がそのまま使えることを確認します。セルを実行すると、計算式と結果が表示されます。

#### 解説: Python の計算

Python では、数式に近い形で計算を書けます。

- `+`: 足し算
- `-`: 引き算
- `*`: 掛け算
- `/`: 割り算（結果は小数になります。例: `8 / 2` は `4.0`）
- `**`: べき乗（例: `2 ** 3` は $2^3 = 8$）

**ポイント:**

- 掛け算は `×` ではなく `*`、割り算は `÷` ではなく `/` を使います
- `//` を使うと「割り算の整数部分だけ」を取り出せます（例: `7 // 2` は `3`）
- `%` を使うと「割り算の余り」を取り出せます（例: `7 % 2` は `1`）


In [ ]:
# ===== Step 0: 加減乗除 =====
# TODO: まずはセルを実行して、計算結果が下に表示されることを確認する
print('1 + 2 =', 1 + 2)
print('5 - 3 =', 5 - 3)
print('4 * 6 =', 4 * 6)
print('8 / 2 =', 8 / 2)
print('2 ** 3 =', 2 ** 3)  # べき乗: 2の3乗

# シャープ（#）はコメントアウトの記号です。コードの実行には影響しません。
# +, -, *, / はそれぞれ足し算、引き算、掛け算、割り算です
# ** はべき乗（何乗するか）を表します
# print() は、() の中の値を出力する関数です。''の中の文字列はそのまま出力されます。,の後の値は計算されて出力されます。

1 + 2 = 3
5 - 3 = 2
4 * 6 = 24
8 / 2 = 4.0


### Step 1: 変数を使って計算する

値を変数に入れてから、変数同士の計算を確認します。数字を変えると結果も変わるので、セルを少し書き換えて試してください。

#### 解説: 変数とは

変数は、値に名前を付けて保存する箱のようなものです。一度変数に入れておけば、後でその名前を使って値を再利用できます。

$$
a = 10,\quad b = 4 \implies a + b = 14
$$

例えば、次のように考えられます。

- `a = 10`: 「`a` という箱に `10` を入れる」という意味です
- `a + b`: 箱の中身同士を足します（`10 + 4 = 14`）

変数を使うと、数字を変えたいときに1箇所だけ直せば良くなります。

**ポイント:**

- `=` は「右の値を左の変数に入れる」操作です（数学の等号とは少し違います）。数学の「左と右が等しい」という意味ではないことに注意しましょう
- 変数名は半角英数字で、先頭は文字または `_` にします（例: `price`, `total_1`）。数字から始まる名前は使えません（例: `1price` はNG）
- 変数は何度でも上書きできます（例: `a = 10` の後で `a = 20` とすると、`a` は `20` になります）
- 変数名は「何が入っているか分かりやすい名前」をつけるのが良い慣習です（例: 面積なら `area`、横幅なら `width`）

次のセルでは、`a` や `b` の値を変えて、計算結果がどう変わるか試してください。

In [2]:
# ===== Step 1: 変数 =====
# TODO: 数字を入れ替えて、結果がどう変わるか確認する
a = 10
b = 4

print('a =', a)
print('b =', b)
print('a + b =', a + b)
print('a - b =', a - b)
print('a * b =', a * b)
print('a / b =', a / b)

# ヒント: = は値を変数に入れる記号です

a = 10
b = 4
a + b = 14
a - b = 6
a * b = 40
a / b = 2.5



例えば、長方形の面積は次のように考えられます。

$$
面積 = 横幅 \times 高さ
$$

コードでは `width * height` のように書きます。

In [3]:
# ===== 追加演習: 面積を計算する =====
# TODO: width と height の値を変えて、面積がどう変わるか確認する
width = 3
height = 5
area = width * height

print('横幅 =', width)
print('高さ =', height)
print('面積 =', area)


横幅 = 3
高さ = 5
面積 = 15


### Step 2: 文字列・リスト・条件分岐・繰り返し

数値の計算だけでなく、文字や複数の値も扱ってみます。ここでは「読みながら動かす」ことを重視してください。

#### 解説: 文字列

文字列は、文字の並びを `' '` や `" "` で囲んで表します。

- `'Iris'`: データセットの名前
- `'cm'`: 単位

変数に文字列を入れることもできます（例: `name = 'Iris'`）。

**ポイント:**

- `'` と `"` はどちらでも使えますが、前後で揃える必要があります（例: `'Iris"` はエラーになります）
- 文字列同士は `+` で連結できます（例: `'Iris' + ' setosa'` → `'Iris setosa'`）

#### 解説: リスト

リストは、複数の値を順番に並べたものです。角括弧 `[ ]` で囲み、要素を `,` で区切ります。

- `features = ['sepal length', 'sepal width', 'petal length', 'petal width']`
- `len(features)`: リストの長さ（要素数）を確認
- `features[0]`: 1つ目の要素にアクセス（Python では番号は0から始まります）
- `features[-1]`: 最後尾の要素にアクセス

**ポイント:**

- リストの番号は0から始まるので、`features[0]` が最初、`features[1]` が2番目になります。これは「先頭からの距離」と考えると分かりやすいです（先頭は0番目、1つ先は1番目）
- `-1` を使うと最後の要素、`-2` で後ろから2番目にアクセスできます（例: 4つの要素がある場合、`features[-1]` は4つ目、`features[-2]` は3つ目）
- `append()` でリストの最後に新しい要素を追加できます（例: `features.append('species')` で `'species'` が5つ目の要素として追加されます）
- リストには文字列だけでなく、数値も混ぜて入れることができます（例: `[1, 'two', 3]`）

次のセルで文字列とリストを試してください。

In [4]:
# ===== Step 2-1: 文字列とリスト =====
name = 'Iris'
unit = 'cm'

print('データ名:', name)
print('単位:', unit)

features = ['sepal length', 'sepal width', 'petal length', 'petal width']
print('特徴量一覧:', features)
print('特徴量の数:', len(features))
print('最初の特徴量:', features[0])

# TODO: 下の1行を完成させて、最後の特徴量を表示する
print('最後の特徴量:', features[-1])

# TODO: append を使って 'species' を追加してみる
features.append('species')
print('追加後の特徴量一覧:', features)

データ名: Iris
単位: cm
特徴量一覧: ['sepal length', 'sepal width', 'petal length', 'petal width']
特徴量の数: 4
最初の特徴量: sepal length
最後の特徴量: petal width
追加後の特徴量一覧: ['sepal length', 'sepal width', 'petal length', 'petal width', 'species']


#### 解説: 条件分岐 (`if`)

条件によって処理を分けるときに `if` を使います。

- `score >= threshold`: 「score が threshold 以上か」を調べます
- 条件が真（True）なら `if` の中のブロックが実行されます
- 偽（False）なら `else` の中のブロックが実行されます

**ポイント:**

- `if` や `else` の行の最後には `:` （コロン）が必要です。これを忘れると `SyntaxError` になります
- ブロックの中は字下げ（インデント）して書きます。Python では、このインデントで「どこからどこまでが `if` の中か」を判断します。通常は半角スペース4個を使います
- 比較には次の記号を使います: `>=`（以上）、`<=`（以下）、`>`（より大きい）、`<`（より小さい）、`==`（等しい）、`!=`（等しくない）
- `==` と `=` は違います。`=` は代入（変数に値を入れる）、`==` は比較（等しいかどうか調べる）です

次のセルで条件分岐を試してください。

In [5]:
# ===== Step 2-2: 条件分岐 =====
score = 75
threshold = 60

print('score =', score)
print('threshold =', threshold)

if score >= threshold:
    print(score, 'は合否ライン', threshold, '以上です')
else:
    print(score, 'は合否ライン', threshold, '未満です')

# TODO: score または threshold を変えて、表示がどう変わるか確認する

score = 75
threshold = 60
75 は合否ライン 60 以上です


#### 解説: 繰り返し (`for`)

リストの要素を1つずつ処理するときに `for` を使います。

- `for fruit in fruits:`: リスト `fruits` の要素を順番に `fruit` に入れて処理します
- `enumerate(fruits)`: 番号（index）と値の両方を同時に取得できます。`for index, fruit in enumerate(fruits):` と書くと、`index` に0からの番号、`fruit` に値が入ります
- `range(n)`: 0 から n-1 までの整数を順番に生成します。`range(start, stop, step)` で開始値・終了値・ステップ幅を指定できます

**ポイント:**

- `for` の行の最後には `:` （コロン）が必要です。`if` と同じです
- ブロックの中身はインデントして書きます
- ループは、リストの要素がなくなるまで自動的に繰り返されます
- ブロックの中身は、各要素に対して1回ずつ実行されます
- `range` を使うと、特定の回数だけループを回すことができます


In [6]:
# ===== Step 2-2: for ループ（基本）=====
fruits = ['apple', 'banana', 'cherry']

for fruit in fruits:
    print(fruit)

# TODO: fruits の中身を変えて、ループの動きを確認する

apple
banana
cherry


In [7]:
# ===== Step 2-3: for ループ =====
fruits = ['apple', 'banana', 'cherry']

for index, fruit in enumerate(fruits):
    print(index, ':', fruit)

# TODO: fruits の中身を変えて、ループの動きを確認する

0 : apple
1 : banana
2 : cherry


In [ ]:
# ===== Step 2-4: range を使った for ループ =====
# range(n): 0 から n-1 までの整数を生成
print("--- range(5) ---")
for i in range(5):
    print(i, end=' ')  # end=' ' は改行の代わりに空白を1つ入れる指定です
print()  # 最後に改行だけ出力する

# range(start, stop): start から stop-1 まで
print("--- range(1, 6) ---")
for i in range(1, 6):
    print(i, end=' ')
print()

# range(start, stop, step): step ずつ増やす
print("--- range(0, 10, 2) ---")
for i in range(0, 10, 2):
    print(i, end=' ')
print()

# TODO: range の引数を変えて、出力がどう変わるか確認する
# ヒント: print(i, end=' ') の end=' ' を end=', ' に変えると、区切り文字が変わります

--- range(5) ---
0 1 2 3 4 
--- range(1, 6) ---
1 2 3 4 5 
--- range(0, 10, 2) ---
0 2 4 6 8 



**発展: データ分析では for ループを避ける理由**

データ分析では、`for` ループで1つずつ計算すると非常に遅くなります。その代わりに、NumPy や pandas の「ベクトル化」機能を使って、配列全体を一度に処理します。

例えば、100万個の値にそれぞれ 2 を掛ける場合:

- `for` ループ: 約 0.1 秒
- NumPy ベクトル化: 約 0.001 秒（100倍速い）

この授業では、Step 4 以降で NumPy のベクトル化に触れます。まずは `for` の仕組みを理解し、その後で「なぜベクトル化が速いか」を体験してください。

次のセルで for ループを試してください。

### Step 3: 関数を作る

同じ計算を何度も使いたいときは、関数にまとめると便利です。

関数では、
- 入力: 関数に渡す値（引数と呼びます）
- 出力: 関数が返す値（戻り値と呼びます）

まずは長方形の面積を求める関数を読んで、入力を変えて試してください。

In [ ]:
# ===== Step 3: 関数 =====
# def で関数を定義します。関数名(引数1, 引数2, ...) と書きます
# return で結果（戻り値）を返します
def rectangle_area(width, height):
    return width * height

print('3 x 5 の面積 =', rectangle_area(3, 5))
print('7 x 2 の面積 =', rectangle_area(7, 2))

# TODO: 引数を変えて、面積がどう変わるか確認する
# ヒント: rectangle_area(10, 20) のように、好きな数字を入れてみてください

3 x 5 の面積 = 15
7 x 2 の面積 = 14


### Step 4: NumPy で配列と行列を扱う

データ分析では、数値が並んだ配列をまとめて扱うことが多くあります。

`NumPy` は、数値計算を高速に行うための基本ライブラリです。Python のリストと比べて、次のような特徴があります。

- **高速**: C 言語で実装されており、大量の数値を一度に処理できます
- **簡潔**: 配列全体に対する計算が、1 行のコードで書けます
- **行列計算**: 2 次元配列を「行列」として扱い、行列演算が容易です

ここでは 2 次元配列を「ベクトルの集まり、つまり行列」として見ます。さらに高次元配列は、「行列の集まり」として捉えることができます。

この段階では、次の操作を確認できれば十分です。

- 配列の形（shape）を確認する
- 行ごと、列ごとに平均を計算する
- 配列同士の四則演算を行う

平均以外にも、積和・内積・転置など、様々な行列演算の機能を使うことができます。

**用語の確認:**

- **配列（array）**: NumPy で数値を並べたもの。Python のリストに似ていますが、数値計算に特化しています
- **shape**: 配列の形を表します。2次元配列の場合は `(行数, 列数)` で表示されます
- **axis**: 平均などを計算するときの「方向」を指定します。`axis=0` は列方向（各列の平均）、`axis=1` は行方向（各行の平均）です

In [ ]:
# ===== Step 4: NumPy 配列 =====
import numpy as np  # NumPy を読み込み、np という短い名前で使えるようにする

scores = np.array([
    [80, 65, 70],
    [90, 88, 92],
    [75, 60, 78]
])

print('scores =')
print(scores)
print('形 =', scores.shape)  # (行数, 列数) = (3, 3)
print('各行の平均 =', scores.mean(axis=1))  # 行方向（横）の平均
print('各列の平均 =', scores.mean(axis=0))  # 列方向（縦）の平均

# NumPy の強み: 配列全体を一度に計算できる
print('全要素に 10 点加算 =')
print(scores + 10)
print('全要素を 2 倍 =')
print(scores * 2)

# TODO: 1つ値を変えて、平均がどう変わるか確認する
# TODO: scores + 10 の結果が、元の scores を書き換えないことを確認する
#       （再度 print(scores) を実行して、元の値のままであることを確かめてください）

scores =
[[80 65 70]
 [90 88 92]
 [75 60 78]]
形 = (3, 3)
各行の平均 = [71.66666667 90.         71.        ]
各列の平均 = [81.66666667 71.         80.        ]
全要素に 10 点加算 =
[[ 90  75  80]
 [100  98 102]
 [ 85  70  88]]
全要素を 2 倍 =
[[160 130 140]
 [180 176 184]
 [150 120 156]]


### Step 5: ライブラリを読み込んでデータを準備する

ここからは、次回以降の分析で使う表形式データに少し触れます。この回の中心は Python の基本操作なので、グラフはまだ描きません。

Python自体では、高度な数式処理、表の処理、グラフの描画の機能を提供していません。それらを行うためには、自分で機能を実装する必要があります。しかし、あらゆる機能を自分で実装するのは現実的ではありません。そこで、Pythonにない便利な機能をまとめた「道具箱」のようなもの、**ライブラリ**を使い、様々な機能を簡単に使えるようにします。ライブラリを読み込むことで、複雑な処理を短いコードで実行できるようになります。

データ分析では、主に次の 3 つのライブラリを使います。

- **NumPy**: 数値計算・配列処理の基本ライブラリ
- **pandas**: 表形式データを扱うライブラリ。Excel のような行・列構造でデータを操作できます
- **scikit-learn (sklearn)**: 機械学習のライブラリ。データ読み込み、前処理、モデルの学習・評価までを一通り提供します

**ライブラリの読み込み方:**

`import` を使ってライブラリを読み込みます。よく使うライブラリには短い別名（エイリアス）をつける慣習があります。

- `import numpy as np`: NumPy を `np` という名前で使う
- `import pandas as pd`: pandas を `pd` という名前で使う
- `from sklearn.datasets import load_iris`: scikit-learn の中から `load_iris` という関数だけを取り出す

このステップでは、scikit-learn に同梱されている Iris データセットを読み込み、pandas の DataFrame として準備します。

#### 解説: 表形式データと DataFrame

Iris データセットは、アヤメの花の特徴を 150 件収録した表形式データです。機械学習の練習で最もよく使われるデータセットの一つです。

- 1行: 1つの花の観測データ
- 1列: 花びらの長さ、がく片の幅などの特徴量
- `species`: 花の種類（3 品種のいずれか）

**DataFrame とは**

pandas の `DataFrame` は、行と列で構成される表形式のデータ構造です。Excel のシートに似ており、次のような操作が容易に行えます。

- 列の追加・削除
- 条件に合う行の抽出
- 集計（平均、合計、件数など）

データ分析では、モデルを作る前にまず次を確認します。

- 行数と列数
- 列名
- 先頭数行
- 値の範囲や平均

これは「データが何を表しているかを理解する」ためであり、分析の最初のステップとして重要です。


#### 補足: Iris データセットの3品種

Iris データセットに収録されている花は、すべてアヤメ属（Iris属）の花です。次の3品種がそれぞれ50件ずつ、合計150件収録されています。

| `target` | 品種名 | 日本語名 | 特徴 |
|----------|--------|----------|------|
| 0 | setosa | ヒオウギアヤメ | 花びらが短く幅も狭い。他の2品種と明確に分離しやすい |
| 1 | versicolor | ブルーフラッグ | setosa と virginica の中間的な大きさ |
| 2 | virginica | バージニカアヤメ | 花びらが長く幅も広い傾向がある |

**ポイント:**

- `target` 列には 0, 1, 2 という数値が入っています。これは上の表の順番に対応しています
- 次のコードセルで `species` 列を作るとき、この数値を品種名（`'setosa'` など）に変換しています
- 機械学習では、4つの特徴量（がく片・花びらの長さと幅）から「この花がどの品種か」を当てる（分類する）練習によく使われます
- 次回の可視化で、品種ごとに特徴量の分布がどう違うかをグラフで確認する予定です



In [ ]:
# ===== Step 5: ライブラリ読み込みとデータ準備 =====
import pandas as pd
from sklearn.datasets import load_iris

# scikit-learn に同梱されている Iris データセットを読み込む
iris = load_iris(as_frame=True)

# DataFrame としてコピーする
df = iris.frame.copy()

# target (数値) を species (品種名) に変換する
# target は 0, 1, 2 の数値で品種を表しています。これを 'setosa' などの名前に変換します
df['species'] = df['target'].map(dict(enumerate(iris.target_names)))

print('行数, 列数 =', df.shape)
display(df.head())  # 先頭5行を表形式で表示する

# ヒント: DataFrame は表のようにデータを扱うための型です
#       load_iris() は scikit-learn に同梱された練習用データです
#       display() は Jupyter で表をきれいに表示する関数です（print() より見やすい形式になります）

行数, 列数 = (150, 6)


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target,species
0,5.1,3.5,1.4,0.2,0,setosa
1,4.9,3.0,1.4,0.2,0,setosa
2,4.7,3.2,1.3,0.2,0,setosa
3,4.6,3.1,1.5,0.2,0,setosa
4,5.0,3.6,1.4,0.2,0,setosa


### Step 6: データを確認する

列名、基本統計量、件数を確認します。

この回では「表として何が入っているか」を確認するところまでに留めます。

分布や関係をグラフで見る作業はテーマ2で行います。

**使うメソッドの確認:**

- `df.columns`: 列名の一覧（「何のデータが入っているか」を確認する最初のステップ）
- `df.describe()`: 各列の基本統計量（平均、標準偏差、最小値、最大値など）を一覧表示
- `df['species'].value_counts()`: 指定した列の値ごとに件数を集計（この場合、品種ごとに何件あるか）

In [ ]:
# ===== Step 6: データ確認 =====
# TODO: 先頭5行、列名、基本統計量を確認する
print('列名一覧')
print(df.columns.tolist())  # 列名をリストとして表示
print('\n基本統計量')
display(df.describe())  # 平均・標準偏差・最小値・最大値など

print('\n品種ごとの件数')
display(df['species'].value_counts())  # 品種ごとに件数を集計

# ヒント: `df['列名']` で1つの列を取り出せます
#       df.describe() で各列の平均、標準偏差、最小値、最大値などが確認できます
#       value_counts() は、同じ値がいくつあるかを数える便利なメソッドです
#       \n は print の中で「改行」を表す特別な文字です


列名一覧
['sepal length (cm)', 'sepal width (cm)', 'petal length (cm)', 'petal width (cm)', 'target', 'species']

基本統計量


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),target
count,150.000000,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333,1.000000
std,0.828066,0.435866,1.765298,0.762238,0.819232
min,4.300000,2.000000,1.000000,0.100000,0.000000
25%,5.100000,2.800000,1.600000,0.300000,0.000000
50%,5.800000,3.000000,4.350000,1.300000,1.000000
75%,6.400000,3.300000,5.100000,1.800000,2.000000
max,7.900000,4.400000,6.900000,2.500000,2.000000



品種ごとの件数


species
setosa        50
versicolor    50
virginica     50
Name: count, dtype: int64

#### ミニ演習: データから1文で説明する

下の文を自分の観察に合わせて書き換えてください。

正解を探すよりも、「どの出力を根拠にそう言ったのか」を意識して説明することが大切です。

例
- 基本統計量から、`________` の平均は `________` あたりだとわかる。
- 品種ごとの件数を見ると、`________` が `________` 件ある。
- 列名から、このデータは `________` に関する表だと考えられる。

## 振り返り
- 加減乗除と変数のどこで意味が分かりにくかったですか。
- 文字列・リスト・条件分岐・繰り返しのどの部分が新しく感じましたか。
- 関数は「何を入力して、何を返すもの」だと理解しましたか。
- NumPy の2次元配列では、行と列をどう読むと理解しやすかったですか。
- Iris データにはどのような列があり、それぞれ何を表していそうですか。
- 次回の可視化で確かめてみたいことを1つ書いてください。

## 演習課題

以下の演習をノートブック上で実行し、結果を確認してください。

### 演習1: 基本計算と変数 (簡単)

1. 変数 `x` に好きな数字を代入し、それを2倍した値を表示してください。
2. 変数 `price` に 1500 を代入し、8%の消費税を加えた合計金額を計算して表示してください。

```python
x =           # ここで数値を代入します。
print("2倍:", )

price = 1500
tax_rate = 0.08
total = 
print("税込価格:", total)
```

### 演習2: リストと条件分岐 (簡単)

1. 変数 `scores` に `[75, 88, 62, 95, 43]` を代入してください。
2. このリストの各要素について、60以上なら「合格」、それ以下なら「不合格」と表示するコードを書いてください。

```python
# ここにコードを書いてください
scores = [75, 88, 62, 95, 43]
for score in scores:
    if :
        print(f"{score}点 → ")
    else:
        print(f"{score}点 → ")
```

### 演習3: Irisデータ確認 (簡単)

1. `iris` データセットの列名を再度確認し、それぞれの列が何を表しているか1行ずつ書いてください。
2. `df.describe()` の結果から、`sepal length (cm)` の最大値と最小値を読み取り、その差（範囲）を計算してください。

```python
# ここにコードを書いてください
print("列名:", )

desc = df.describe()
sepal_max = desc.loc['max', 'sepal length (cm)']
sepal_min = desc.loc['min', 'sepal length (cm)']
print(f"sepal length (cm) の範囲: {sepal_max - sepal_min}")
```

---

## レポート課題

以下のレポートを提出してください（A4 1ページ程度、またはノートブックのセルに記述）。

### レポート項目

1. **Pythonの基本操作** (200字程度)
   - 変数・リスト・条件分岐・繰り返しの中で、最も新しく感じた概念はどれですか。その理由を説明してください。

2. **NumPy配列** (200字程度)
   - 1次元配列と2次元配列の違いを、自分の言葉で説明してください。また、2次元配列の「行」と「列」はどのように区別しましたか。

3. **Irisデータセット** (200字程度)
   - Irisデータにはどのような列がありますか。それぞれの列が何を表しているか、そしてこのデータを使って何ができるかを説明してください。

4. **次回への期待** (100字程度)
   - 次回の「データ可視化」で、特に確認してみたいことは何ですか。

